src/v/wasm/allocator.cc

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "wasm/allocator.h"
+
+#include <seastar/core/align.hh>
+#include <seastar/core/aligned_buffer.hh>
+#include <seastar/core/print.hh>
+
+#include <stdexcept>
+#include <unistd.h>
+
+namespace wasm {
+namespace {
+bool is_aligned(size_t v, size_t alignment) {
+    return ss::align_up(v, alignment) == v;
+}
+} // namespace
+
+heap_allocator::heap_allocator(config c) {
+    size_t page_size = ::getpagesize();
+    if (!is_aligned(c.heap_memory_size, page_size)) {
```

Comment:
any reason not to do `heap_memory_size = align_up(heap_memory_size, page_size)` or something like that?

Reply:
Nope. Done.